### ISSUE TITLE
Fails to open build shell after build failure

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#346]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/346)
In GitLab by [[Gitlab user @TheRealMichaelCatanzaro]](https://gitlab.com/TheRealMichaelCatanzaro) on Apr 8, 2018, 18:29

I have a local workspace open for WebKit, and have modified it such that it fails to build. BuildStream gives me an interactive prompt to decide what to do, and I decide I want a shell:

```
Build failure on element: core-deps/WebKit.bst

Choose one of the following options:
  (c)ontinue  - Continue queueing jobs as much as possible
  (q)uit      - Exit after all ongoing jobs complete
  (t)erminate - Terminate any ongoing jobs and exit
  (r)etry     - Retry this job
  (l)og       - View the full log file
  (s)hell     - Drop into a shell in the failed build sandbox

Pressing ^C will terminate jobs and exit

Choice: [continue]: s
```

Problem is it opens a runtime shell, not a build shell. I have no access to the build directory and no ability to do anything useful inside the shell. Surely that's not correct?

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 8, 2018, 19:06

Hmmm seems like a regression. We don't have good regression tests for the interactive parts :-/

I'll have to take a look and and reproduce when it's daytime... it's *not* trying to setup a runtime shell, and it should be using the failed build sandbox directly, according to [the code](https://gitlab.com/BuildStream/buildstream/blob/master/buildstream/_frontend/app.py#L632)

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @TheRealMichaelCatanzaro]](https://gitlab.com/TheRealMichaelCatanzaro) on Apr 8, 2018, 19:25

It might be the build sandbox without the builddir present. I just notice that there's not any files inside the sandbox directory.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 8, 2018, 19:29

That may be part of the reason behind regression; this is part of work towards making incremental builds work well for open workspaces: The workspace directory gets mounted into the sandbox shell instead of doing copies.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 11, 2018, 10:11

mentioned in commit 6076969ae979a348abece53fdb26c7ab3c0db085

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 11, 2018, 10:11

mentioned in commit 9bd57475945b562b6582e1aa82c5508b9b2d5a98

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 11, 2018, 10:43

mentioned in commit d46fbbcbfec7f44cfa3d412b70000eff633b85bb

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 11, 2018, 10:43

mentioned in commit a1564c4a1cf077b1a8537d37e19d1748a0905b20

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 11, 2018, 10:54

closed via merge request !380

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Apr 11, 2018, 10:54

closed via commit d46fbbcbfec7f44cfa3d412b70000eff633b85bb

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Jul 23, 2018, 12:01

mentioned in merge request !504
